File: src/main/java/it/contrader/dto/ThingDTO.java
package it.contrader.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ThingDTO {
	private Integer idThing;
	private String code;
	private String description;
	private String image;
	private String name;
	private String xml;
	private String protocol;

	private UserDTO user;
	private LabelDTO label;
	private List<KeywordDTO> keyword;
	
}

File: IncAng/src/dto/ThingDTO.ts
import { UserDTO } from './UserDTO';
import { LabelDTO } from './LabelDTO';

export class ThingDTO{
    idThing: number;
    code: String;
    description: String;
    image: String;
    name: String;
    xml: String;
    protocol: String;
    user: UserDTO;
    label: LabelDTO;

    constructor(id:number, code: String, description: String, image: String, name: String, xml: String, protocol: String, userDTO: UserDTO, labelDTO: LabelDTO){
        this.idThing=id;
        this.code=code;
        this.description=description;
        this.image=image;
        this.name=name;
        this.xml=xml;
        this.protocol=protocol;
        this.user=userDTO;
        this.label=labelDTO;
    }
}